SU-SWS/stanford_basic
# D8CORE-931: Skip links

# READY FOR REVIEW 

# Summary
- Generates unique anchor tags for skipping to content
- The primary goal is to have a `main-content` id anchor to target no matter where and how the templates are used. 

# TODO
- Unit Tests
- Behat Tests

# Review By (Date)
- Jan 31 (Friday)

# Urgency
- Would be nice to have by the end of the sprint

# Steps to Test

1. Check out this branch
2. Check out the `skips` branch of jumpstart_ui
3. Require `drupal/twig_extender` in your root composer.json
4. Do a site install of stanford_profile or run database updates if you already have one installed
5. Create a nested menu item page if you don't already have one so that secondary navigation shows up in the sidebar
6. Navigate to the home page
7. Reload the page and hit tab. The skip to main-content option should show up and take you to the main content
8. Reload the page and hit tab twice. No secondary navigation skip link should be available
9. Navigate to `/search/content` and hit tab
10. The skip to main content option should be available but should be linking to `#page-content` instead of `main-content`
11. Navigate to the page with the secondary nav
12. Hit tab until you see the skip to secondary navigation option and hit return
13. You should be at the secondary navigation item now
14. Reload the page
15. Hit tab until you see an option to skip to main content just before the secondary navgation

# Affected Projects or Products
- D8CORE

# Associated Issues and/or People
- D8CORE-931
- D8CORE-909
- D8CORE-930

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
